react/bingo/src/bingo/GameLogic.js

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import './BingoApp.css';
+import { cleanup } from '@testing-library/react';
+
+class GameLogic extends React.Component {
+    constructor(props) {
+        super(props);
+        this.maskedChars = [];
+
+        this.state = {
+            attemptLetter: '',
+            failedLetters: '',
+            successLetters: '',
+            challengeWord: [],
+            maskedWord: [],
+            isGameOver : false,
+            bingoColor: [
+                'attempt-left',
+                'attempt-left',
+                'attempt-left',
+                'attempt-left',
+                'attempt-left'
+            ]
+        };
+
+    }
+    getBlankText = () => {
+        let challengeText = this.props.challenge;
+        let size = challengeText.length;
+        let challengeChars = challengeText.split('');
+        let isInit = true;
+        let maskedChars = [];
+        if (isInit) {
+            isInit = false;
+            maskedChars = [];
+            for (let i = 0; i < size; i++) {
+                maskedChars.push('*');
+            }
+        } else {
+            maskedChars = this.state.maskedWord;
+        }
+        this.maskedChars =  (this.state.maskedWord.length === 0 
+            ? maskedChars : this.state.maskedWord);
+
+        return (
+            <div className='board-row'>
+                <label className={this.state.bingoColor[0]} >B</label>
+                <label className={this.state.bingoColor[1]} >I</label>
+                <label className={this.state.bingoColor[2]} >N</label>
+                <label className={this.state.bingoColor[3]} >G</label>
+                <label className={this.state.bingoColor[4]} >O</label>
+                <br></br>
+                {this.maskedChars.map((chara, i) =>
+                    <RenderChallenge
+                        guessedchar={chara}
+                        visible='true' />)} 
+                <br />
+                <h4>Wrongly Guessed Letters </h4>
+                <label className='bingo-wrong-answer'>
+                    {this.state.failedLetters.toUpperCase()}
+                </label>
+
+            </div>
+        );
+    }
+
+    handleSubmit = () => {
+        if (this.state.isGameOver !== true) {
+            if (this.state.successLetters.includes(this.state.attemptLetter) 
```

Comment:
use "lodash" library for most of the trivial string, array operations

Reply:
Will incorporate the change for the new project. Closing the comment